src/accounting.rs

```diff
@@ -1,56 +1,84 @@
-use crate::constants::{
-    CANISTER_OVERHEAD, COLLATERAL_CYCLES_PER_NODE, HTTP_OUTCALL_REQUEST_BASE_COST,
-    HTTP_OUTCALL_REQUEST_COST_PER_BYTE, HTTP_OUTCALL_REQUEST_PER_NODE_COST,
-    HTTP_OUTCALL_RESPONSE_COST_PER_BYTE, INGRESS_MESSAGE_BYTE_RECEIVED_COST,
-    INGRESS_MESSAGE_RECEIVED_COST, INGRESS_OVERHEAD_BYTES, NODES_IN_SUBNET, RPC_URL_COST_BYTES,
-};
+use crate::constants::{COLLATERAL_CYCLES_PER_NODE, INGRESS_OVERHEAD_BYTES, RPC_URL_COST_BYTES};
 
 /// Calculates the cost of sending a JSON-RPC request using HTTP outcalls.
-pub fn get_http_request_cost(payload_size_bytes: u64, max_response_bytes: u64) -> u128 {
-    let nodes_in_subnet = NODES_IN_SUBNET as u128;
-    let ingress_bytes =
+/// See https://internetcomputer.org/docs/current/developer-docs/gas-cost/#https-outcalls
+pub fn get_http_request_cost(
+    nodes_in_subnet: u32,
+    payload_size_bytes: u64,
+    max_response_bytes: u64,
+) -> u128 {
+    let n = nodes_in_subnet as u128;
+    let request_bytes =
         payload_size_bytes as u128 + RPC_URL_COST_BYTES as u128 + INGRESS_OVERHEAD_BYTES;
```

Comment:
You probably forgot to include the ingress message cost here. Here’s an example of how we implemented cost calculation: https://github.com/mfactory-lab/ic-solana/blob/3ed64a636b3441ea3b3417169ac012b94c3c3f15/src/ic-solana-rpc/src/http.rs#L69

Reply:
Thanks for the suggestion! The code you quoted was exactly what this PR is trying to change, because that was an old formula that is no longer in use now (but is probably safe because it overestimates over 20%).

Reply:
Is ingress fee does not use anymore ?

Reply:
The correct documentation for the cost calculation is this one https://internetcomputer.org/docs/current/developer-docs/gas-cost/#https-outcalls

We are aware that there was another document that had a wrong formula https://internetcomputer.org/docs/current/references/https-outcalls-how-it-works/#pricing, which needs to be amended.

Neither document mentioned ingress fee. In fact, since https outcalls are made from canisters, there is no ingress message involved at all. So at the very least any cost involved shouldn't be called "ingress fee".

Reply:
Thanks @ninegua